com.unity.visualeffectgraph/Editor/ShaderGraph/VFXShaderGraphParticleOutput.cs

```diff
@@ -880,6 +868,25 @@ public override void EndCompilation()
                             }
                         }
 
+                        if (graphCode.requirements.requiresScreenPosition
+                            || graphCode.requirements.requiresNDCPosition
+                            || graphCode.requirements.requiresPixelPosition)
+                        {
+                            callSG.builder.AppendLine("{");
+                            if (graphCode.requirements.requiresNDCPosition || graphCode.requirements.requiresPixelPosition)
+                                callSG.builder.AppendLine("float2 localPixelPosition = i.VFX_VARYING_POSCS.xy;");
+                            if (graphCode.requirements.requiresNDCPosition)
+                                callSG.builder.AppendLine("float2 localNDCPosition = localPixelPosition.xy / _ScreenParams.w;");
```

Comment:
Does this make sense? It the variable even used and what does it represents?

Reply:
I bet it's a typo between .w and .xy. But still, this line seems uneeded as ndc is computed a few lines below

Reply:
I removed the localNDCPosition & localPixelPosition for code generation (localPixelPosition == i.VFX_VARYING_POSCS.xy;). It prevents a missing condition and improve readability.

Reply:
I also quickly tested all cases from ScreenPosition matching between VFX & SG: 
https://user-images.githubusercontent.com/22494706/134672858-30a2a786-7d90-46fd-a273-706be48b6114.mp4

Reply:
However, @cdxntchou About this https://github.com/Unity-Technologies/Graphics/blob/75f4b421fa774d43000d703f2bbf05d7a1ca6606/com.unity.render-pipelines.universal/Editor/ShaderGraph/Templates/SharedCode.template.hlsl#L56

Do you know why `ScreenPosition` (so, if I understand correctly the exposed "Raw" settings from the ScreenPosition node in ShaderGraph) always uses a TransformWorldToHClip(worldPos) ?
Can't we recompute the initial clip position using SV_Position (multiplying by w to return into the initial homogenous space) ? I think I'm missing an obvious constraint which prevent us to remove this matrix mulplication in pixel.

Reply:
N.B.: We discussed internally about this possibility, it's a potential improvement but we should check the consistency among platform first.
See this [conversation](https://unity.slack.com/archives/C02G8EDKLLQ/p1632505097008800)